lib/runtime-core/src/typed_func.rs

```diff
@@ -124,16 +124,49 @@ pub trait WasmTypeList {
         self,
         f: NonNull<vm::Func>,
         wasm: Wasm,
-        ctx: *mut Ctx,
+        ctx: *mut vm::Ctx,
     ) -> Result<Rets, RuntimeError>
     where
         Rets: WasmTypeList;
 }
 
+/// Empty trait to specify the kind of `ExternalFunction`: With or
+/// without a `vm::Ctx` argument. See the `ExplicitVmCtx` and the
+/// `ImplicitVmCtx` structures.
+///
+/// This type is never aimed to be used by a user. It is used by the
+/// trait system to automatically generate an appropriate `wrap`
+/// function.
+pub trait ExternalFunctionKind {}
```

Comment:
Can it be made `pub (crate)`?

Reply:
Nah unfortunately, since it's used by `ExternalFunction` which is public.